src/drivers/express/ExpressAdminRouteDriver.ts

```
@@ -80,24 +84,30 @@ export class ExpressAdminRouteDriver {
         } else {
           learningObjects = await AdminLearningObjectInteractor.fetchAllObjects(
             this.dataStore,
+            accessGroups,
             page,
             limit,
           );
         }
         res.status(200).send(learningObjects);
       } catch (e) {
         console.error(e);
+        if (e instanceof Error) {
```

I really like that we are adding some better error handling logic here 👍. Is there a way we could ensure that if there is an Error thrown, it is in fact for invalid access? Currently, this will catch all errors and so I could see us getting some weird responses in certain failure situations.

Perhaps we could throw the access error with a particular string as its message: 
```
throw new Error('Invalid Access');
```

So that we could do something like:
```
if (e instanceof Error && e.message === 'Invalid Access')
```

---

Ok, I like this. I was trying to avoid string comparison, but I see your point. After reviewing your refactor-to-modules PR, I noticed the LearningObjectError const. Maybe we can store all error strings in there?

---

Yeah, having a constant that is used when throwing and evaluating the error would definitely be a better way to go with this.